pennylane_lightning/lightning_qubit/_state_vector.py

```diff
@@ -0,0 +1,349 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Class implementation for state-vector manipulation.
+"""
+
+try:
+    from pennylane_lightning.lightning_qubit_ops import (
+        StateVectorC64,
+        StateVectorC128,
+        allocate_aligned_array,
+        best_alignment,
+        get_alignment,
+    )
+except ImportError:
+    pass
+
+from itertools import product
+
+import numpy as np
+import pennylane as qml
+from pennylane import BasisState, DeviceError, StatePrep
+from pennylane.tape import QuantumScript
+from pennylane.wires import Wires
+
+
+class LightningStateVector:
+    """Lightning state-vector class.
+
+    Interfaces with C++ python binding methods for state-vector manipulation.
+
+    Args:
+        num_wires(int): the number of wires to initialize the device with
+        dtype: Datatypes for state-vector representation. Must be one of
+            ``np.complex64`` or ``np.complex128``.
+        device_name(string): state vector device name. Options: ["lightning.qubit"]
+    """
+
+    def __init__(self, num_wires, dtype=np.complex128, device_name="lightning.qubit"):
+        self._num_wires = num_wires
+        self._wires = Wires(range(num_wires))
+        self._dtype = dtype
+
+        if dtype not in [np.complex64, np.complex128]:  # pragma: no cover
+            raise TypeError(f"Unsupported complex type: {dtype}")
+
+        if device_name != "lightning.qubit":
+            raise DeviceError(f'The device name "{device_name}" is not a valid option.')
+
+        self._device_name = device_name
+        self._qubit_state = self._state_dtype()(self._num_wires)
+
+    @property
+    def dtype(self):
+        """Returns the state vector data type."""
+        return self._dtype
+
+    @property
+    def device_name(self):
+        """Returns the state vector device name."""
+        return self._device_name
+
+    @property
+    def wires(self):
+        """All wires that can be addressed on this device"""
+        return self._wires
+
+    @property
+    def num_wires(self):
+        """Number of wires addressed on this device"""
+        return self._num_wires
+
+    def _state_dtype(self):
+        """Binding to Lightning Managed state vector C++ class.
+
+        Returns: the state vector class
+        """
+        return StateVectorC128 if self.dtype == np.complex128 else StateVectorC64
+
+    @staticmethod
+    def _asarray(arr, dtype=None):
```

Comment:
I was afraid the functionality could be needed in the future.
But if you think it is a good idea to remove it, I will do it.

Reply:
Yep, please! if this is not required for the current support. let's remove it so that we always add tested features and helper methods to the codebase.